### ISSUE TITLE
V10 - SendingContentNotification does not work as in V9

### ISSUE THREAD

**Issue by olibos** [umbraco/Umbraco-CMS#12586]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

10.0.0

### Bug summary

Hello, 

I think there is an issue introduce by the following PR: #11398 
As `ContentItemDisplayWithSchedule` does not inherit from `ContentItemDisplay` it breaks the notifications sent by: `[OutgoingEditorModelEvent]`.

Is this intentional? And if yes is there any alternatives?

### Specifics

Original code : 
https://github.com/umbraco/Umbraco-CMS/blob/48003e0b707499f39116aad2a2f148e931a70049/src/Umbraco.Web.BackOffice/Controllers/ContentController.cs#L322-L333

New Code:
https://github.com/umbraco/Umbraco-CMS/blob/eccdb36bc0f5a0d225ced546c3fa5eca6bbb6fd7/src/Umbraco.Web.BackOffice/Controllers/ContentController.cs#L322-L333

Event handling:
https://github.com/umbraco/Umbraco-CMS/blob/060e0181a784c03057d76f508397bfc9684fd3e2/src/Umbraco.Web.BackOffice/Filters/OutgoingEditorModelEventAttribute.cs#L70-L74

### Steps to reproduce

* Create a simple Notification Handler for `SendingContentNotification`:
```c#
builder.AddNotificationHandler<SendingContentNotification, DisablePreviewNotificationHandler>();

public class DisablePreviewNotificationHandler: INotificationHandler<SendingContentNotification>
{
    public void Handle(SendingContentNotification notification)
    {
        notification.Content.AllowPreview = false;
    }
}
````
* Go on any content, and you will see that the `Save and preview` is shown. 😒

### Expected result / actual result

Expected resullt (like in V9):
![image](https://user-images.githubusercontent.com/6031263/174087017-6f1e1445-e9b7-4599-90c2-9c82f3ef7cd1.png)

Actual result in V10:
![image](https://user-images.githubusercontent.com/6031263/174087112-2e32a4dc-3d33-42bd-b2aa-53489c327033.png)

**Comment by nathanwoulfe** [comment]
Hey @olibos I can reproduce this one, and agree with your diagnosis 😄 I can see how it might have been assumed to work and equally how the variations of `ContentItemX` model names quickly gets a bit confusing and convoluted.

As a quick verification, I created an `IContentItemDisplay` interface, updated `ContentItemDisplay` and `ContentItemDisplayWithSchedule` to both implement that interface, then updated the outgoing event filter and content notification to use the interface instead of the concrete type. With that in place the notification is handled correctly.

IMO, given there are now different type of content display model, there should be an interface to describe the implementations. Not sure though what the HQ preference would be.

Either way, this is a regression, and will be looked after by the good folk at HQ.

**Comment by nikolajlauridsen** [comment]
Just had a look at this yesterday, and you're both entirely right (Thanks for making my job easy 💪 😄)

Unfortunately though, we can create and interface and use that in the notification since it would be a breaking change, so long story short what I'd have to do is map between the two different displays, which seems to work quite well :) 

I'll close this however since it's fixed in #12597
